ou_dedetai/tui_app.py

```diff
@@ -23,109 +32,198 @@
-        self.main_window_ratio = self.main_window_ratio = self.menu_window_ratio = self.main_window_min = None
-        self.menu_window_min = self.main_window_height = self.menu_window_height = self.main_window = None
-        self.menu_window = self.resize_window = None
+        self.tui_screens: list[tui_screen.Screen] = []
+        self.menu_options: list[Any] = []
+
+        self._installer_thread: Optional[threading.Thread] = None
+
+        # Default height and width to something reasonable so these values are always
+        # ints, on each loop these values will be updated to their real values
+        self.window_height = self.window_width = 80
+        self.main_window_height = self.menu_window_height = 80
+        # Default to a value to allow for int type
+        self.main_window_min: int = 0
+        self.menu_window_min: int = 0
+
+        self.menu_window_ratio: Optional[float] = None
+        self.main_window_ratio: Optional[float] = None
+        self.main_window_ratio = None
+        self.main_window: Optional[curses.window] = None
+        self.menu_window: Optional[curses.window] = None
+        self.resize_window: Optional[curses.window] = None
+
+        # For menu dialogs.
+        # a new MenuDialog is created every loop, so we can't store it there.
+        self.current_option: int = 0
+        self.current_page: int = 0
+        self.total_pages: int = 0
+
+        # Start internal property variables, shouldn't be accessed directly, see their 
+        # corresponding @property functions
+        self._menu_screen: Optional[tui_screen.MenuScreen] = None
+        self._active_screen: Optional[tui_screen.Screen] = None
+        self._console: Optional[tui_screen.ConsoleScreen] = None
+        # End internal property values
+
+        # Lines for the on-screen console log
+        self.console_log: list[str] = []
+
+        # Note to reviewers:
+        # This does expose some possibly untested code paths
+        #
+        # Before some function calls didn't pass use_python_dialog falling back to False
+        # now it all respects use_python_dialog
+        # some menus may open in dialog that didn't before.
+        self.use_python_dialog: bool = False
+        if "dialog" in sys.modules and ephemeral_config.terminal_app_prefer_dialog is not False: #noqa: E501
+            result = system.test_dialog_version()
+
+            if result is None:
+                logging.debug(
+                    "The 'dialog' package was not found. Falling back to Python Curses."
+                )  # noqa: E501
+            elif result:
+                logging.debug("Dialog version is up-to-date.")
+                self.use_python_dialog = True
+            else:
+                logging.error(
+                    "Dialog version is outdated. The program will fall back to Curses."
+                )  # noqa: E501
+        # FIXME: remove this hard-coding after considering whether we want to continue 
```

Comment:
It's a good question. The main reason, as explained in another comment, was for TUI file/directory selection.